src/Symfony/Component/Mailer/Bridge/Amazon/Transport/RequestSignTrait.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Amazon\Transport;
+
+use Symfony\Component\Mailer\Bridge\Amazon\Credential\ApiTokenCredential;
+use Symfony\Component\Mailer\Bridge\Amazon\Credential\UsernamePasswordCredential;
+
+/**
+ * @author Karoly Gossler
+ */
+trait RequestSignTrait
+{
+    private function getSignatureHeaders(): array
+    {
+        $headers = [];
+        $date = gmdate('D, d M Y H:i:s e');
+
+        if ($this->credential instanceof ApiTokenCredential) {
+            $accessKey = $this->credential->getAccessKey();
+            $secretKey = $this->credential->getSecretKey();
+
+            $headers['X-Amz-Security-Token'] = $this->credential->getToken();
+        } elseif ($this->credential instanceof UsernamePasswordCredential) {
+            $accessKey = $this->credential->getUsername();
+            $secretKey = $this->credential->getPassword();
+        }
+
+        $signature = base64_encode(hash_hmac('sha256', $date, $secretKey, true));
```

Comment:
this is using the old version 3 signature scheme, which is not even described in the AWS doc anymore (the SES doc says it supports it, but does not describe the algorithm anymore). Please use the version 4 signature scheme instead.